src/index.ts

```diff
@@ -65,98 +79,50 @@ export function composeStory<GenericArgs>(
   meta: Meta<GenericArgs | any>,
   globalConfig: GlobalConfig = globalStorybookConfig
 ) {
-
-  if (isInvalidStory(story)) {
-    throw new Error(
-      `Cannot compose story due to invalid format. @storybook/testing-react expected a function/object but received ${typeof story} instead.`
-    );
-  }
-
-  if (story.story !== undefined) {
-  throw new Error(
-      `StoryFn.story object-style annotation is not supported. @storybook/testing-react expects hoisted CSF stories.
-       https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#hoisted-csf-annotations`
-    );
+  if(story === undefined) {
+    throw new Error('Expected a story but received undefined.')
   }
 
-  const renderFn = typeof story === 'function' ?  story : story.render ?? globalRender;
-  const finalStoryFn = (context: StoryContext<ReactFramework, GenericArgs>) => {
-    const { passArgsFirst = true } = context.parameters;
-    if (!passArgsFirst) {
-      throw new Error(
-        'composeStory does not support legacy style stories (with passArgsFirst = false).'
-      );
-    }
-
-    return renderFn(context.args, context);
-  };
-
-  const combinedDecorators = [
-    ...(story.decorators || []),
-    ...(meta?.decorators || []),
-    ...(globalConfig.decorators || []),
-  ];
-
-  const decorated = defaultDecorateStory<ReactFramework>(
-    finalStoryFn as any,
-    combinedDecorators as any
+  const projectAnnotations = { ...defaultGlobalConfig, ...globalConfig }
+  
+  // @TODO: instead of processCSFFile this should be 
+  // normalizeStories + normalizeComponentAnnotations from @storybook/store
+  const { 
+    meta: normalizedMeta, 
+    stories: normalizedStories
+  } = processCSFFile({ default: meta, ...{[story.storyName!]: story} }, '', meta.title)
+  const normalizedStory = Object.values(normalizedStories)[0] as any
+
+  const preparedStory = prepareStory(
+    normalizedStory,
+    normalizedMeta,
+    projectAnnotations
   );
 
   const defaultGlobals = Object.entries(
```

Comment:
Hey @tmeasday I'm sure there's an existing way to compose the globalTypes in Storybook. Do you know where? Something related with the urlStore?

Reply:
Here! https://github.com/storybookjs/storybook/blob/4e007aa29b89d7cfd1da03479b6407ff008ccedb/lib/store/src/GlobalsStore.ts#L22-L37